### ISSUE TITLE
Not able to reproduce the mAP results after training

### ISSUE THREAD

**Issue by gurkirt** [eriklindernoren/PyTorch-YOLOv3#193]
Thank a lot for producing this amazing code! 

I am training to train yolov3 on coco
After preprocessing the dataset, I ran

`` python3 train.py --data_config config/coco.data  --pretrained_weights weights/darknet53.conv.74 ``

Got mAp of 20.1 % after 70 epochs

`` python3 test.py --weights_path weights/yolov3.weights `` 

This gave me mAp of 51.4% rather than 55.5%. 

Do I need to specify some other parameters?

**Comment by Invokar** [comment]
@gurkirt  Hi! I'm a little confuse that where yolov3.weight from? because when I use yolov3.weight to test on COCO, it's reuslt is really good! But If I train 
`python3 test.py --weights_path weights/yolov3.weights`
The first epoch the result is really bad? why?

**Comment by yanxurui** [comment]
Maybe the current implementation of yolo-v3 using pytorch is not trainable?
It's only useful for inference when using the pretrained weights by the official darknet.

**Comment by CodePothunter** [comment]
“This gave me mAp of 51.4% rather than 55.5%.”

Same issue, but I don't know why.

**Comment by gurkirt** [comment]
@eriklindernoren 
Is this version issue? I am using pytorch1.1

**Comment by ktian08** [comment]
I'm also running the default training instructions per the README and I'm only getting ~22 mAP after 100 epochs.

**Comment by ktian08** [comment]
I ran the default training command as well, and only got to ~22 mAP after 100 epochs. The losses are plateau'd.

**Comment by fmahoudeau** [comment]
A bug has been found in collate_fn which impacts the evaluation: https://github.com/eriklindernoren/PyTorch-YOLOv3/issues/243

After implementing the fix the mAP increases to 54.9%.

**Comment by www12345678** [comment]
@fmahoudeau   
Is your map tested on voc data or coco data sets?

**Comment by www12345678** [comment]
@fmahoudeau 
Is your mAP tested on voc data or coco data sets?

**Comment by www12345678** [comment]
@fmahoudeau 
Is your mAP tested on voc data or coco data sets?

**Comment by fmahoudeau** [comment]
> @fmahoudeau
> Is your mAP tested on voc data or coco data sets?

It was tested on COCO.

**Comment by www12345678** [comment]
@fmahoudeau
When you tested on coco,how big do you choose epochs?

**Comment by www12345678** [comment]
@fmahoudeau
When you tested on coco,how big do you choose epochs? chose yolov3.weights or darknet53.conv.74 to training?

**Comment by z-huabao** [comment]
> @gurkirt Hi! I'm a little confuse that where yolov3.weight from? because when I use yolov3.weight to test on COCO, it's reuslt is really good! But If I train
> `python3 test.py --weights_path weights/yolov3.weights`
> The first epoch the result is really bad? why?

Hi, I test `yolov3.weights` get mAP=54.8, and get 33 after fine-tune 1 epoch by 1e-6 learning rate.

**Comment by YuanduLai** [comment]
> Thank a lot for producing this amazing code!
> 
> I am training to train yolov3 on coco
> After preprocessing the dataset, I ran
> 
> `python3 train.py --data_config config/coco.data --pretrained_weights weights/darknet53.conv.74`
> 
> Got mAp of 20.1 % after 70 epochs
> 
> `python3 test.py --weights_path weights/yolov3.weights`
> 
> This gave me mAp of 51.4% rather than 55.5%.
> 
> Do I need to specify some other parameters?

do you solve this problem now?

**Comment by BaiqiangGit** [comment]
I get mAP: 0.5492690730408117

**Comment by falex-ml** [comment]
> Thank a lot for producing this amazing code!
> 
> I am training to train yolov3 on coco
> After preprocessing the dataset, I ran
> 
> `python3 train.py --data_config config/coco.data --pretrained_weights weights/darknet53.conv.74`
> 
> Got mAp of 20.1 % after 70 epochs
> 
> `python3 test.py --weights_path weights/yolov3.weights`
> 
> This gave me mAp of 51.4% rather than 55.5%.
> 
> Do I need to specify some other parameters?

Same here

**Comment by gurkirt** [comment]
Try what https://github.com/eriklindernoren/PyTorch-YOLOv3/issues/193#issuecomment-515774443 says

**Comment by ujsyehao** [comment]
When I use #243 proposed method , mAP increase from 51.4% to 54.9%, but still lower than 55.3%.

Another problem is TP calculation in utils/utils.py, please refer to #233

**Comment by zhangyilalala** [comment]
python3 train.py --data_config config/coco.data --pretrained_weights weights/darknet53.conv.74

I still get mAp of 19 % after 60 epochs after fix the code in #243, did anyone train well on coco?

**Comment by WangJian981002** [comment]
> “This gave me mAp of 51.4% rather than 55.5%.”
> 
> Same issue, but I don't know why.

the confidence in inference is set to high, you can change it to 0.15
